docs/reference/language/context.md

```diff
@@ -25,38 +25,180 @@ in some places that are also aware of their location in the document:
 [Show rules]($styling/#show-rules) provide context[^1] and numberings in the
 outline, for instance, also provide the proper context to resolve counters.
 
-## Style context
-With set rules, we can adjust style properties for parts or the whole of our
-document. We cannot access these without a known context, as they may change
-throughout the course of the document. When context is available, we can
-retrieve them simply by accessing them as fields on the respective element
-function.
+## Behavior of the context keyword
+Style properties frequently change within a document, for example by applying set 
+rules. To retrieve such poperties in a consistent way, one must first specify 
+the context where the query is to be executed. This is the purpose of the 
+`context` keyword. Once the context has been fixed, the property information 
+is available through a simple field access syntax. For example, `text.lang` 
+asks for the current language setting. In its simplest form, the `context` 
+keyword refers to "right here":
 
 ```example
 #set text(lang: "de")
+// query the language setting "here"
 #context text.lang
 ```
 
-As explained above, a context expression is reactive to the different
-environments it is placed into. In the example below, we create a single context
-expression, store it in the `value` variable and use it multiple times. Each use
-properly reacts to the current surroundings.
+Note that calling `#text.lang` directly would be an error, because the request
+cannot be answered without knowledge of the context. The field names supported 
+by a given element function always correspond to the named parameters documented 
+on each element's page.
+
+Moreover, some functions, such as [`to-absolute`]($length.to-absolute) 
+and [`counter.display`]($counter.display) are only applicable in a context, 
+because their results depend on the current settings of style properties. 
+When another function `foo()` calls a context-dependent function, it becomes 
+itself context-dependent:
+
+```example
+#let foo() = 1em.to-absolute()
+#context {
+  // foo() cannot be called outside of a context
+  foo() == text.size
+}
+```
+
+When a property is changed, the response to the query changes accordingly: 
 
 ```example
-#let value = context text.lang
-#value
+#set text(lang: "en")
+#context text.lang
 
 #set text(lang: "de")
-#value
+#context text.lang
+```
+
+The output of a `#context ...` call is static in the form of opaque 
+[`content`]. Write access to context output is prohibited, as it would 
+often result in invalid code: If the context changes between read and 
+write, overwriting a property would cause an inconsistent system state. 
```

Comment:
This doesn't read quite correctly. The crucial thing is that contextual content might appears 0, 1, or N times in the document. There is some more discussion of this in the second part of this blog post: https://laurmaedje.github.io/posts/types-and-context/

Reply:
I'm not happy with this sentence either. However, I would like to have a one-sentence explanation why things need to be more complicated than what one would naively expect. Talking about "closures" etc. as in your post doesn't feel right, because many users won't know this term. Any suggestions?